qiskit/pulse/channels/pulse_specification.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""
+Specification of the device.
+"""
+from typing import List
+
+from .channels import AcquireChannel, MemorySlot, RegisterSlot
+from .pulse_channels import DriveChannel, ControlChannel, MeasureChannel
+from .qubit import Qubit
+
+
+class PulseSpecification:
+    """Implement a device specification."""
```

Comment:
Could you please provide a more detailed summary and description of the purpose/use case of this class?